contrib/testgen/gen_base58_test_vectors.py

```diff
@@ -96,20 +222,11 @@ def gen_invalid_vectors():
     # start with some manual edge-cases
     yield "",
     yield "x",
+    glist = [gen_invalid_base58_vector, gen_invalid_bech32_vector]
+    tlist = [templates, bech32_ng_templates]
     while True:
-        # kinds of invalid vectors:
-        #   invalid prefix
-        #   invalid payload length
-        #   invalid (randomized) suffix (add random data)
-        #   corrupt checksum
-        for template in templates:
-            val = gen_invalid_vector(template, randbool(0.2), randbool(0.2), randbool(0.2))
-            if random.randint(0,10)<1: # line corruption
-                if randbool(): # add random character to end
-                    val += random.choice(b58chars)
-                else: # replace random character in the middle
-                    n = random.randint(0, len(val))
-                    val = val[0:n] + random.choice(b58chars) + val[n+1:]
+        for template, gen_invalid_vector in [(t, g) for g, l in zip(glist, tlist) for t in l]:
+            val = gen_invalid_vector(template)
```

Comment:
A bit long, but maybe call it `invalid_vector_generator`, since it is very close to the function names above. I thought for a moment it was referring to the function called `gen_invalid_vectors` but it had an `s`, I realized.